### PR TITLE
POST/add book

### LINKED ISSUE #39: POST/ book route
#### book  Route (appointment)
POST new appointment 

--

#### Route :
`api/v1/book
`
--

#### Method : POST 

---
#### Response : 
Response may be success or error :
* Error :
```javascript
       {
        status:400,
        messgae:'Bad reqeust'
    }
```
* Success
```javascript
    {
        status:200,
        message:"booked successfully"
    }
```

### PR DESCRIPTION
**Pull request by AdhamHaisami**
Fix #39 
# Server feature

 - Create AddBook route,
 -  Build two needed DB queries
    * checkAvailability
    * addBook
 -  Build controller funcion.
 -  Test route and queries.

### CODE REVIEW

**Review by imansedky5 — DISMISSED** [review]

File: __test__/addBook.test.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  describe('Test addBook route in failur case', () => {
+    const failerBookingData = {
+      userId: 2,
+      serviceId: 2,
+      stylistId: 2,
+      appoinmtmentDate: '2021-03-28',
+      appointmentTime: '12:00',
+    };
+    test('Expect to return object with status 400, and success message', async () => {
```
Comment (mFlaifel) [inline __test__/addBook.test.js:74]: please fix typos
image
- install code spell checker to detects any typos

File: __test__/addBook.test.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      const {
+        body: { statusCode, errorMessage },
+      } = await request(app)
+        .post('/api/v1/booking')
+        .set({ 'Content-Type': 'application/json' })
+        .send({ ...failerBookingData });
+
+      expect(statusCode).toBe(400);
+      expect(errorMessage).toBe('choosen time in not available');
```
Comment (mFlaifel) [inline __test__/addBook.test.js:83]:
```suggestion
      const {
        body: { statusCode, message },
      } = await request(app)
        .post('/api/v1/booking')
        .set({ 'Content-Type': 'application/json' })
        .send({ ...failerBookingData });

      expect(statusCode).toBe(400);
      expect(message).toBe('choosen time in not available');
```

File: server/controller/user/addBook.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      appoinmtmentDate,
+      appointmentTime,
+    } = req.body;
+    const { rows } = await checkAvailability(appoinmtmentDate, appointmentTime);
+    if (rows.length === 0) {
+      await addBook({
+        userId,
+        serviceId,
+        stylistId,
+        appoinmtmentDate,
```
Comment (mFlaifel) [inline server/controller/user/addBook.js:20]: more typos
image

File: server/database/queries/user/addBook.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  userId,
+  serviceId,
+  stylistId,
+  appoinmtmentDate,
+  appointmentTime,
+}) => {
+  const sql = {
+    text:
+      'INSERT INTO appointments (user_id, service_id, stylist_id, appointment_date, appointment_time) VALUES ($1,$2,$3,$4,$5) RETURNING appointment_date, appointment_time ',
+    values: [userId, serviceId, stylistId, appoinmtmentDate, appointmentTime],
```
Comment (mFlaifel) [inline server/database/queries/user/addBook.js:13]: maybe returning all data will be helpful for the frontend developer.
```suggestion
    text:
      'INSERT INTO appointments (user_id, service_id, stylist_id, appointment_date, appointment_time) VALUES ($1,$2,$3,$4,$5) RETURNING * ',
    values: [userId, serviceId, stylistId, appoinmtmentDate, appointmentTime],
```

File: server/controller/user/addBook.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+const addNewBook = async (req, res, next) => {
+  try {
+    const {
+      userId,
+      serviceId,
+      stylistId,
+      appoinmtmentDate,
+      appointmentTime,
+    } = req.body;
```
Comment (mFlaifel) [inline server/controller/user/addBook.js:13]: Suggestion,
if you have time, you can add validate the request body

File: server/database/config/build.sql [hunk 6]
```diff
@@ -59,6 +59,7 @@ CREATE TABLE appointments (
   "appointment_time" VARCHAR(100) NOT NULL
 );
 
+
```
Comment (mFlaifel) [inline server/database/config/build.sql:62]: Tell me why?

**Pull request merged**